compiler/rustc_traits/src/check_instance_wf.rs

```diff
@@ -0,0 +1,28 @@
+use rustc_infer::infer::TyCtxtInferExt as _;
+use rustc_middle::ty::{self, TyCtxt, TypingEnv};
+use rustc_trait_selection::error_reporting::InferCtxtErrorExt;
+use rustc_trait_selection::traits::{Obligation, ObligationCause, ObligationCtxt};
+
+pub(crate) fn check_instance_wf<'tcx>(tcx: TyCtxt<'tcx>, instance: ty::Instance<'tcx>) {
```

Comment:
isn't that just what `instantiate_and_check_impossible_predicates` does?

https://github.com/rust-lang/rust/blob/ad211ced81509462cdfe4c29ed10f97279a0acae/compiler/rustc_trait_selection/src/traits/mod.rs#L701-L742

Reply:
i don't really want feedback on this pr, i'm just using it for crater.